test/rematch2.jl

```diff
@@ -227,25 +300,25 @@ end
 end
 
 @testset "test for sharing where clause disjuncts" begin
-    # State 1 TEST «input_value» isa Main.Rematch2Tests.Foo ELSE: State 18 («label_2»)
-    # State 2 FETCH «input_value.x» := «input_value».x
-    # State 3 FETCH «input_value.y» := «input_value».y
-    # State 4 TEST «input_value.y» == 2 ELSE: State 11 («label_3»)
-    # State 5 FETCH «where_0» := (f1)((identity)(«input_value.x»))
-    # State 6 TEST !«where_0» ELSE: State 8 («label_5»)
-    # State 7 MATCH 1 with value 1
-    # State 8 («label_4») TEST «input_value.x» == 1 THEN: State 10 ELSE: State 18 («label_2»)
-    # State 9 («label_5») TEST «input_value.x» == 1 ELSE: State 13 («label_3»)
-    # State 10 FETCH «where_1» := (f2)(«input_value.y»)
-    # State 11 TEST where !«where_1» ELSE: State 18 («label_2»)
-    # State 12 MATCH 2 with value 2
-    # State 13 («label_3») FETCH «where_0» := (f1)(«input_value.x»)
-    # State 14 TEST where !«where_0» ELSE: State 18 («label_2»)
-    # State 15 FETCH «where_1» := (f2)(«input_value.y»)
-    # State 16 TEST where !«where_1» ELSE: State 18 («label_2»)
-    # State 17 MATCH 3 with value 3
-    # State 18 («label_2») MATCH 4 with value 5
-    @test (Rematch2.@match2_count_states some_value begin
+    # Node 1 TEST «input_value» isa Main.Rematch2Tests.Foo ELSE: Node 18 («label_2»)
+    # Node 2 FETCH «input_value.x» := «input_value».x
+    # Node 3 FETCH «input_value.y» := «input_value».y
+    # Node 4 TEST «input_value.y» == 2 ELSE: Node 11 («label_3»)
+    # Node 5 FETCH «where_0» := (f1)((identity)(«input_value.x»))
+    # Node 6 TEST !«where_0» ELSE: Node 8 («label_5»)
+    # Node 7 MATCH 1 with value 1
+    # Node 8 («label_4») TEST «input_value.x» == 1 THEN: Node 10 ELSE: Node 18 («label_2»)
+    # Node 9 («label_5») TEST «input_value.x» == 1 ELSE: Node 13 («label_3»)
+    # Node 10 FETCH «where_1» := (f2)(«input_value.y»)
+    # Node 11 TEST where !«where_1» ELSE: Node 18 («label_2»)
+    # Node 12 MATCH 2 with value 2
+    # Node 13 («label_3») FETCH «where_0» := (f1)(«input_value.x»)
+    # Node 14 TEST where !«where_0» ELSE: Node 18 («label_2»)
+    # Node 15 FETCH «where_1» := (f2)(«input_value.y»)
+    # Node 16 TEST where !«where_1» ELSE: Node 18 («label_2»)
+    # Node 17 MATCH 3 with value 3
+    # Node 18 («label_2») MATCH 4 with value 5
+    @test (Rematch2.@match2_count_nodes some_value begin
```

Comment:
could we add a (or some) tests similar to this one that check the type propagation and the filtering that allows on the state machine?

Reply:
See line 701 of `rematch2.jl`.

Reply:
at line 701 I see - has this moved?
<img width="605" alt="image" src="https://github.com/JuliaServices/Rematch2.jl/assets/24885580/f611910a-d947-479c-bf35-e2191a123a8a">

Reply:
It is at 717 now.